### PR TITLE
Add shape inference for Ops used by BERT

### PR DESCRIPTION
**Pull request by albertomagni-ms**
* Erf
* Pow
* ReduceMean
* Dropout
* Expand
  https://github.com/onnx/onnx/blob/master/docs/Operators.md#expand
  Deduce the value of the shape operand by looking at the producer
  of the operand.
  Currently supported producers are: onnx.Constant and onnx.Shape.

* Add corresponding tests for each op.

* Sort the list of ops with shape inference in gen_onnx_mlir.py
in alphabetic order for clarity.

### CODE REVIEW

**Comment by albertomagni-ms** [comment]
Hi all, 
please let me know if you would like to split this PR into multiple patches. 
Also, I am happy to add additional tests if needed.

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 1]
```diff
@@ -2655,6 +2691,90 @@ LogicalResult ONNXSliceOp::inferShapes() {
...
+    rhsShape.assign(rhsShapeRef.begin(), rhsShapeRef.end());
+
+  } else if (mlir::ONNXConstantOp constantOp =
+                 dyn_cast_or_null<mlir::ONNXConstantOp>(shapeDef)) {
+    // If the shape operand is produced by a onnx.Constant operation, extract
+    // the actual value of the constant and use it as the reqested shape.
+
+    auto shapeTensorTy = shape().getType().cast<RankedTensorType>();
+
+    if (shapeTensorTy.getShape().size() != 1)
```
[the remark this answers is not in the document]
Comment (albertomagni-ms) [reply src/Dialect/ONNX/ONNXOps.cpp]: Fixed! Thank you.

**Comment by albertomagni-ms** [comment]
The build appears to be flaky:
https://travis-ci.org/github/onnx/onnx-mlir/jobs/715456307
The PPC build failed: An error occurred while generating the build script.

**Comment by tjingrant** [comment]
@albertomagni-ms my guess is because we are currently in a bit of a broken state because we're trying to upgrade the llvm-project commit ID. Will talk to people who have more knowledge about it and will keep you posted!

**Comment by tjingrant** [comment]
@jenkins-droid test this please

**Comment by albertomagni-ms** [comment]
I don't have permission to merge this. Please go ahead when ready :) Thank you.

**Comment by tjingrant** [comment]
@albertomagni-ms Done, thanks!

**Comment by Joejiong** [comment]
is bert being support by the flow: bert.onnx-->bert.mlir -->llvmir --> bitcode --> obj file?

**Pull request merged**